config/modules/dns/master/default.nix

```diff
@@ -0,0 +1,132 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hackint.dns.master;
+
+  zone = pkgs.writeText "${cfg.zones.dynamic.domain}.zone" ''
+    $ORIGIN ${cfg.zones.dynamic.domain}.
+    $TTL 60
+
+    @ SOA ${head cfg.nameservers} mail.hackint.org 0 200 300 1209600 300
+
+    ${concatMapStringsSep "\n"
+      (server: "@ NS ${server}")
+      cfg.nameservers}
+  '';
+
+in
+{
+  options.hackint.dns.master = with types; {
+    enable = mkEnableOption "the DNS master";
+
+    upstreams = mkOption {
+      type = listOf str;
+      description = ''
+        Remote adrresses of upstream servers.
+      '';
+    };
+
+    nameservers = mkOption {
+      type = listOf str;
+      description = ''
+        FQDNs of public serving servers.
+      '';
+    };
```

Comment:
Semantically this should probably be a per-zone option, even if we reference the same list of nameservers over and over again.

Reply:
Zones are now unified and nameservers and upstream remotes are now per-zone